### PR TITLE
Enable anonymous access

### PR DESCRIPTION
**Pull request by kcasas**
/cc @zendesk/mintegrations

### Description

Enable anonymous access. Added missing return on some functions.

### Risks
[medium] May break functionalities that expects a thrown exception when auth values are not supplied

### CODE REVIEW

File: tests/Zendesk/API/UnitTests/Core/AuthTest.php [hunk 1]
```diff
@@ -12,6 +16,23 @@
 class AuthTest extends BasicTest
 {
     /**
+     * Test if request is still sent even without auth details
+     */
+    public function testAnonymousAccess()
+    {
+        $client = $this
```
Comment (samgavinio) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: This test doesn't seem like it's testing the anonymous access behaviour directly or at least it isn't too obvious. Shouldn't there at least be an assertion that `$client->getAuth()->prepareRequest()` is not called?

Also there's a `client` class property which contains an instance of `Zendesk\API\HttpClient` already, may be you can use that.

Comment (kcasas) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: I believe there is no assertion for method calls. If there is, kindly refer it to me. I've tried using that client but it seems that it is not a mocked object.

Comment (miogalang) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: you can do 

`$client->expects('getAuth')->never()`

File: src/Zendesk/API/Http.php [hunk 2]
```diff
@@ -82,7 +82,10 @@ public static function send(
         }
 
         try {
-            list ($request, $requestOptions) = $client->getAuth()->prepareRequest($request, $requestOptions);
+            // enable anonymous access
+            if ($client->getAuth()) {
+                list ($request, $requestOptions) = $client->getAuth()->prepareRequest($request, $requestOptions);
```
Comment (joseconsador) [inline src/Zendesk/API/Http.php]: `getAuth()` doesn't really do much and is not reused in the library. Perhaps we can move the implementation of this conditional to `Zendesk\API\Utilities\Auth::prepareRequest()` .

File: tests/Zendesk/API/UnitTests/Core/AuthTest.php (hunk 1 above)
Comment (kcasas) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: getAuth is called once when checking if it exists or not.

Comment (samgavinio) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: 👍 on the client object not being a mocked object. I guess you can create your own mock but you should be able to assert the method call if you use mockery demeter chains:

http://docs.mockery.io/en/latest/reference/demeter_chains.html

Comment (miogalang) [inline tests/Zendesk/API/UnitTests/Core/AuthTest.php]: Maybe prepareRequest is what you should check as not being called then.

File: src/Zendesk/API/Http.php (hunk 2 above)
Comment (joseconsador) [inline src/Zendesk/API/Http.php]: That's possible, the Auth class would no longer be needed if that's the case. Check out the [ruby implementation](https://github.com/zendesk/zendesk_api_client_rb/blob/master/lib/zendesk_api/client.rb#L158) for some inspiration.

**Pull request merged**